### PR TITLE
Allowing constant objective function

### LINKED ISSUE #79: constant objective function
The [line](https://github.com/StructJuMP/StructJuMP.jl/blob/master/src/StructJuMP.jl#L258) does not allow to have a constant objective function. We can have the following argument type:
```
::Union{Real, AbstractJuMPScalar}
```

### PR DESCRIPTION
**Pull request by kibaekkim**

### CODE REVIEW

**Comment by kibaekkim** [comment]
It is strange that `Real <: AbstractJuMPScalar` returns `false`. This could have been addressed in `JuMP`. But, this commit would resolve that.

**Comment by blegat** [comment]
`Real` is not an `AbstractJuMPScalar` as it does not corresponds to any MOI scalar function.
We should rather do as in https://github.com/JuliaOpt/JuMP.jl/pull/2247

**Comment by kibaekkim** [comment]
> `Real` is not an `AbstractJuMPScalar` as it does not corresponds to any MOI scalar function.
> We should rather do as in [jump-dev/JuMP.jl#2247](https://github.com/jump-dev/JuMP.jl/pull/2247)

@blegat Can you elaborate a bit more about your suggestion? Is this comment on https://github.com/StructJuMP/StructJuMP.jl/pull/80/commits/b25afc99ea311bc79536a6be6923c1190ef06185? 
My fix was base on https://github.com/jump-dev/JuMP.jl/blob/release-0.21/src/objective.jl#L107-L114

**Comment by blegat** [comment]
Add a method
```julia
function JuMP.set_objective_function(m::StructuredModel, f::Real)
    JuMP.set_objective_function(m, convert(AffExpr, f))
end
```
It should work once https://github.com/jump-dev/JuMP.jl/pull/2247 is merged. We should do a patch release of JuMP including this PR soon so that won't block this PR for long

**Comment by kibaekkim** [comment]
@blegat can you check if the changes are compatible to the PR?

File: src/StructJuMP.jl [hunk 1]
```diff
@@ -254,11 +254,17 @@ function JuMP.constraint_object(cref::StructuredConstraintRef, F::Type, S::Type)
...
-function JuMP.set_objective(m::StructuredModel, sense::MOI.OptimizationSense,
-                            f::JuMP.AbstractJuMPScalar)
-    m.objective_sense = sense
+function JuMP.set_objective_function(m::StructuredModel, f::AbstractJuMPScalar)
     m.objective_function = f
 end
+
+JuMP.set_objective_function(m::StructuredModel, f::Real) = JuMP.set_objective_function(m, convert(AffExpr, f))
+
+function JuMP.set_objective(m::StructuredModel, sense::MOI.OptimizationSense, f)
```
Comment (blegat) [inline src/StructJuMP.jl:263]: I would rather implement `JuMP.set_objective_sense(m::StructuredModel, sense::MOI.OptimizationSense)`

Comment (kibaekkim) [reply src/StructJuMP.jl:268]: yeah, make sense.

File: src/StructJuMP.jl [hunk 2]
```diff
@@ -254,11 +254,22 @@ function JuMP.constraint_object(cref::StructuredConstraintRef, F::Type, S::Type)
...
     m.objective_sense = sense
+end
+
+function JuMP.set_objective_function(m::StructuredModel, f::AbstractJuMPScalar)
     m.objective_function = f
 end
+
+JuMP.set_objective_function(m::StructuredModel, f::Real) = JuMP.set_objective_function(m, convert(AffExpr, f))
+
+function JuMP.set_objective(m::StructuredModel, sense::MOI.OptimizationSense, f)
```
Comment (blegat) [inline src/StructJuMP.jl:268]: We don't need that as it's the same as the default in `JuMP` for `AbstractModel`

**Comment by blegat** [comment]
Add in `Project.toml` that we require at least JuMP v0.21.3

File: src/StructJuMP.jl (hunk 2 above)
Comment (kibaekkim) [reply src/StructJuMP.jl:268]: We actually need this function, as `AbstractModel` does not seem to be a superset of `StructuredModel`.

Comment (kibaekkim) [reply src/StructJuMP.jl]: @blegat sorry I was testing with the older JuMP.. The function is removed now.

**Comment by kibaekkim** [comment]
@blegat any other concerns or suggestions?

**Pull request merged**